master/internal/api_trials.go

```diff
@@ -1407,6 +1407,11 @@ func checkpointV2FromProtoWithDefaults(p *checkpointv1.Checkpoint) (*model.Check
 		p.ReportTime = timestamppb.New(time.Now().UTC())
 	}
 
+	var storageID *model.StorageBackendID
+	if p.StorageId != nil {
+		storageID = ptrs.Ptr(model.StorageBackendID(*p.StorageId))
```

Comment:
can we not do
```suggestion
		storageID = (*model.StorageBackendID)(p.StorageId)
```

Reply:
I tried this when I saw this worked in one of Aaron's PRs

But I think the difference here is ```p.StorageId``` is an int32 while model.StorageBackendID is an int so type conversion can't work since the underlying types aren't the same? I'm not sure exactly what the Go language spec rule on this is

Reply:
In depth detail on the rules here   
https://go101.org/article/pointer.html

Reply:
yeah, didn't realize they weren't both ints

Reply:
dang protos